editor/src/clj/editor/settings_core.clj

```diff
@@ -188,9 +214,24 @@
 (defn- category->str [category settings]
   (s/join "\n" (cons (str "[" category "]") (map setting->str settings))))
 
+(defn split-multi-line-setting [category key settings]
+  (let [path [category key]]
+    (if-let [comma-separated-setting (get-setting settings path)]
+      (let [comma-separated-setting-parts (non-blank (s/split comma-separated-setting #","))
+            comma-separated-setting-count (count comma-separated-setting-parts)
+            cleaned-settings (remove-setting settings path)]
+        (reduce
+          (fn [settings i]
+            (set-setting settings [category (str key "#" i)] (nth comma-separated-setting-parts i)))
+          cleaned-settings
+          (range 0 comma-separated-setting-count)))
+      settings)))
+
```

Comment:
I just realized that since we allow users to put their own settings in `game.project`, there is a possibility we will break their stuff if we split every setting that has a `,` in it across multiple lines. I think we might need to only do this for the `dependencies` setting to remain truly backward-compatible.

It also looks like opening a project with no dependencies will now remove the `dependencies = ` line when saved without any modifications to `game.project`. This is obviously a bug that needs to be fixed.

Reply:
It appears I was mistaken about the first part since the `split-multi-line-setting` function only attempts to split a setting if it matches the specified category and key, so ignore that part.